plc4j/spi/src/main/java/org/apache/plc4x/java/spi/Plc4xNettyWrapper.java

```diff
@@ -200,85 +198,60 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
         // by sending a connection request to the plc.
         logger.debug("User Event triggered {}", evt);
         if (evt instanceof ConnectEvent) {
-            this.protocolBase.onConnect(new DefaultConversationContext<>(ctx, authentication, passive));
+            this.protocolBase.onConnect(new DefaultConversationContext<>(this::registerHandler, ctx, authentication, passive));
         } else if (evt instanceof DisconnectEvent) {
-            this.protocolBase.onDisconnect(new DefaultConversationContext<>(ctx, authentication, passive));
+            this.protocolBase.onDisconnect(new DefaultConversationContext<>(this::registerHandler, ctx, authentication, passive));
         } else if (evt instanceof DiscoverEvent) {
-            this.protocolBase.onDiscover(new DefaultConversationContext<>(ctx, authentication, passive));
+            this.protocolBase.onDiscover(new DefaultConversationContext<>(this::registerHandler, ctx, authentication, passive));
         } else if (evt instanceof CloseConnectionEvent) {
-            this.protocolBase.close(new DefaultConversationContext<>(ctx, authentication, passive));
+            this.protocolBase.close(new DefaultConversationContext<>(this::registerHandler, ctx, authentication, passive));
```

Comment:
Here we should stop TimeManager

Reply:
You're right, however I am not entirely sure of netty vs plc4x lifecycle. Namely can we reestablish connection  once we reached that point? At high level plc4x drivers support `connect` call, we would need to assure that it  always configure fresh pipeline and uses new timeout manager forcing netty to fire bootstrap procedure again.

Reply:
Fresh pipeline finnaly always invoke `PlcConnectionManager.getConnection` it will create fesh pipeline. If you don't stop the TimeManager it will caouse memory leak